yt/config.py

```diff
@@ -66,68 +73,142 @@
-    f = open(_OLD_CONFIG_FILE).read()
-    if any(header in f for header in ["[lagos]", "[raven]", "[fido]", "[enki]"]):
-        print("***********************************************************")
-        print("* Upgrading configuration file to new format; saving old. *")
-        print("***********************************************************")
-        # This is of the old format
-        cp = configparser.ConfigParser()
-        cp.read(_OLD_CONFIG_FILE)
-        # NOTE: To avoid having the 'DEFAULT' section here,
-        # we are not passing in ytcfg_defaults to the constructor.
-        new_cp = configparser.ConfigParser()
-        new_cp.add_section("yt")
-        for section in cp.sections():
-            for option in cp.options(section):
-                # We changed them all to lowercase
-                if option.lower() in ytcfg_defaults:
-                    new_cp.set("yt", option, cp.get(section, option))
-                    print(f"Setting {option} to {cp.get(section, option)}")
-        open(_OLD_CONFIG_FILE + ".old", "w").write(f)
-        new_cp.write(open(_OLD_CONFIG_FILE, "w"))
-
-    msg = (
-        "The configuration file {} is deprecated. "
-        "Please migrate your config to {} by running: "
-        "'yt config migrate'"
-    )
-    warnings.warn(msg.format(_OLD_CONFIG_FILE, CURRENT_CONFIG_FILE))
+OLD_CONFIG_FILE = os.path.join(CONFIG_DIR, "ytrc")
+CURRENT_CONFIG_FILE = os.path.join(CONFIG_DIR, "yt.toml")
+
+if os.path.exists(OLD_CONFIG_FILE):
+    if os.path.exists(CURRENT_CONFIG_FILE):
+        msg = (
+            "The configuration file {} is deprecated. "
+            "Please manually remove it to suppress this warning."
+        )
+        warnings.warn(msg.format(OLD_CONFIG_FILE, CURRENT_CONFIG_FILE))
+    else:
+        msg = (
+            "The configuration file {} is deprecated. "
+            "Please migrate your config to {} by running: "
+            "'yt config migrate'"
+        )
+        warnings.warn(msg.format(OLD_CONFIG_FILE, CURRENT_CONFIG_FILE))
+        sys.exit(1)
+
 
 if not os.path.exists(CURRENT_CONFIG_FILE):
-    cp = configparser.ConfigParser()
-    cp.add_section("yt")
+    cfg = {"yt": {}}
     try:
-        with open(CURRENT_CONFIG_FILE, "w") as new_cfg:
-            cp.write(new_cfg)
+        with open(CURRENT_CONFIG_FILE, mode="w") as fd:
+            toml.dump(cfg, fd)
     except IOError:
         warnings.warn("unable to write new config file")
 
 
-class YTConfigParser(configparser.ConfigParser):
-    def __setitem__(self, key, val):
-        self.set(key[0], key[1], val)
+class YTConfig:
+    def __init__(self, defaults=None):
+        if defaults is None:
+            defaults = {"yt": {}}
+        self.defaults = defaults
+        self.values = {}
+        self.update(defaults)
+
+    def get(self, section, *path, strict=True, **kwargs):
+        config = self.values[section]
+
+        # This works as follow: if we try to access
```

Comment:
Does this leave open the opportunity to do frontend-specific config?

Reply:
It does! You can actually do something like 
![image](https://user-images.githubusercontent.com/5411875/100252395-1adc7500-2f40-11eb-9b83-f61c8b940c91.png)